src/ast/mod.rs

```diff
@@ -1210,16 +1172,29 @@ impl fmt::Display for Expr {
             Expr::Array(set) => {
                 write!(f, "{set}")
             }
-            Expr::JsonAccess {
-                left,
-                operator,
-                right,
-            } => {
-                if operator == &JsonOperator::Colon {
-                    write!(f, "{left}{operator}{right}")
-                } else {
-                    write!(f, "{left} {operator} {right}")
+            Expr::VariantAccess { value, path } => {
+                write!(f, "{value}")?;
+                for (i, elem) in path.iter().enumerate() {
```

Comment:
The trouble with that is that an individual `VariantPathElem` needs to know if it's the initial path segment to render as a `:` or a `.` (and I don't want to add `VariantPathElem::Colon` because (a) the two current variants map to the two documented Snowflake syntaxes and (b) it makes more AST states unrepresentable (e.g. if a user manually constructs an AST to render, they have to remember this detail). Another option if we wanted to factor this out would be to wrap the path elements in a `VariantPath` type, though that doesn't really seem worth it to me.